Sources/PostgresNIO/New/Data/Float+PSQLCodable.swift

```diff
@@ -3,18 +3,27 @@ extension Float: PSQLCodable {
         .float4
     }
     
-    static func decode(from buffer: inout ByteBuffer, type: PSQLDataType, context: PSQLDecodingContext) throws -> Float {
-        switch type {
-        case .float4:
+    var psqlFormat: PSQLFormat {
+        .binary
+    }
+    
+    static func decode(from buffer: inout ByteBuffer, type: PSQLDataType, format: PSQLFormat, context: PSQLDecodingContext) throws -> Float {
+        switch (format, type) {
+        case (.binary, .float4):
             guard buffer.readableBytes == 4, let float = buffer.readFloat() else {
                 throw PSQLCastingError.failure(targetType: Self.self, type: type, postgresData: buffer, context: context)
             }
             return float
-        case .float8:
+        case (.binary, .float8):
             guard buffer.readableBytes == 8, let double = buffer.readDouble() else {
                 throw PSQLCastingError.failure(targetType: Self.self, type: type, postgresData: buffer, context: context)
             }
             return Float(double)
```

Comment:
Are we okay with the float-to-double conversion?

Reply:
The implementation I inherited did this, so I guess it's okay for now. Maybe change with the next major release?